cmd/notary/keys.go

```diff
@@ -88,51 +91,117 @@ var cmdKeyImportRoot = &cobra.Command{
+const (
+	maxGUNWidth = 25
+	maxLocWidth = 40
+)
 
-	keys := make(map[trustmanager.KeyStore]map[string]string)
-	for _, store := range stores {
-		keys[store] = store.ListKeys()
-	}
+type keyInfo struct {
+	gun      string // assumption that this is "" if role is root
+	role     string
+	keyID    string
+	location string
+}
 
-	cmd.Println("")
-	cmd.Println("# Root keys: ")
-	for store, keysMap := range keys {
-		for k, v := range keysMap {
-			if v == "root" {
-				cmd.Println(k, "-", store.Name())
-			}
+// We want to sort by gun, then by role, then by keyID, then by location
+// In the case of a root role, then there is no GUN, and a root role comes
+// first.
+type keyInfoSorter []keyInfo
+
+func (k keyInfoSorter) Len() int      { return len(k) }
+func (k keyInfoSorter) Swap(i, j int) { k[i], k[j] = k[j], k[i] }
+func (k keyInfoSorter) Less(i, j int) bool {
+	// special-case role
+	if k[i].role != k[j].role {
+		if k[i].role == data.CanonicalRootRole {
+			return true
 		}
+		if k[j].role == data.CanonicalRootRole {
+			return false
+		}
+		// otherwise, neither of them are root, they're just different, so
+		// go with the traditional sort order.
 	}
 
-	cmd.Println("")
-	cmd.Println("# Signing keys: ")
+	// sort order is GUN, role, keyID, location.
+	orderedI := []string{k[i].gun, k[i].role, k[i].keyID, k[i].location}
+	orderedJ := []string{k[j].gun, k[j].role, k[j].keyID, k[j].location}
 
-	// Get a list of all the keys
-	for store, keysMap := range keys {
-		var sortedKeys []string
-		for k := range keysMap {
-			sortedKeys = append(sortedKeys, k)
+	for x := 0; x < 4; x++ {
+		switch {
+		case orderedI[x] < orderedJ[x]:
+			return true
+		case orderedI[x] > orderedJ[x]:
+			return false
 		}
+		// continue on and evalulate the next item
+	}
+	// this shouldn't happen - that means two values are exactly equal
+	return false
+}
 
-		// Sort the list of all the keys
-		sort.Strings(sortedKeys)
+// Given a list of KeyStores in order of listing preference, pretty-prints the
+// root keys and then the signing keys.
+func prettyPrintKeys(keyStores []trustmanager.KeyStore, writer io.Writer) {
+	var info []keyInfo
 
-		// Print a sorted list of the key/role
-		for _, k := range sortedKeys {
-			if keysMap[k] != "root" {
-				printKey(cmd, k, keysMap[k], store.Name())
+	for _, store := range keyStores {
+		for keyPath, role := range store.ListKeys() {
+			gun := ""
+			if role != data.CanonicalRootRole {
+				gun = filepath.Dir(keyPath)
 			}
+			info = append(info, keyInfo{
+				role:     role,
+				location: store.Name(),
+				gun:      gun,
+				keyID:    filepath.Base(keyPath),
+			})
 		}
 	}
+	sort.Stable(keyInfoSorter(info))
+
+	table := tablewriter.NewWriter(writer)
+	table.SetHeader([]string{"ROLE", "GUN", "KEY ID", "LOCATION"})
+	table.SetBorder(false)
+	table.SetColumnSeparator(" ")
+	table.SetAlignment(tablewriter.ALIGN_LEFT)
+	table.SetCenterSeparator("-")
+	table.SetAutoWrapText(false)
+
+	for _, oneKeyInfo := range info {
```

Comment:
Do you mean instead of the table header, or under it?  E.g:

````
ROLE  GUN ...
-------------
      (no signing keys found)
```

Or just 
```
No signing keys found.
```

?